### DISCUSSION TITLE
How to get .Glb material texture

### DISCUSSION THREAD

**Discussion by base-krchome** [google/model-viewer#2562] [Q&A]
Want to "on"  "off"  buttons for occlusion texture.  But there is a trick in model. The occlusion texture is same with the base color texture.

when the checkbox is "off" i gave it to the value as a white background image so occlusion looks like off. 
When the checkbox is "on" i want to occlusion texture is same as baseColorTexture, but i cant get the material base texture...

```
const modelViewerTexture = document.querySelector("model-viewer#sphere");
  
  modelViewerTexture.addEventListener("load", () => {

  let material = modelViewerTexture.model.materials[0];
 
  let applyTexture = (channel, event) => {
    material[channel].texture.source.setURI(material.pbrMetallicRoughness.baseColorTexture.texture.source);
    }

     
  var checkbox = document.getElementById("ucuncu3");
   
checkbox.addEventListener('click', function() {
      
    applyTexture('occlusionTexture', event);
  });
  });
```

![image](https://user-images.githubusercontent.com/82439010/124839348-4bce5f00-df91-11eb-91cc-28508fd30d76.png)

You guys can show the material's textures on 2D at modelviewer.dev/editor/... So there is a way to reach the material's textures in some ways.
How? Can you help me?

**Comment by shamsail** [comment]
you are doing this wrong way.

```
material[channel].texture.source.setURI(material.pbrMetallicRoughness.baseColorTexture.texture.source);
```
because 
```
material.pbrMetallicRoughness.baseColorTexture.texture.source
```
this is not a URI. here URI is the URL that needs a path to the file. That's why you are getting the error of `[object%20Object]`

There is a function/property
```
material.pbrMetallicRoughness.baseColorTexture.texture.source.uri
```
But, it will give you undefined. It's not working currently. **v-1.7.2**

So, what you can do is to take out the texture image from the 
```
material.pbrMetallicRoughness.baseColorTexture.texture.source
```
and make some computations own your own to achieve your thing.
like, img to uri and use this uri in your `setURI` method.

This is a self hack as currently there is no function or property that will give you the uri/image/base64/blob directly.
May be we will find this feature in next release. Meanwhile you can see this to get the image and make uri out of it.

https://github.com/google/model-viewer/discussions/2531

**Reply by elalish** [reply]
Yes, the URI is null if the texture is embedded, which is standard for GLB. If you use a glTF with separate texture files, you can access the URI as described. We are working on a more flexible texture API, so stay tuned!
